app/src/main/java/com/nutomic/syncthingandroid/service/RestApi.java

```diff
@@ -536,4 +542,23 @@ public void getUsageReport(final OnResultListener1<String> listener) {
     public URL getUrl() {
         return mUrl;
     }
+
+    public Boolean isUsageReportingDecided() {
+        Options options = getOptions();
+        if (options == null) {
+            Log.e(TAG, "isUsageReportingDecided called while options == null");
+            return true;
+        }
+        return options.isUsageReportingDecided(mUrVersionMax);
+    }
+
+    public void setUsageReporting(Boolean acceptUsageReporting) {
+        Options options = getOptions();
+        if (options == null) {
+            Log.e(TAG, "setUsageReporting called while options == null");
+            return;
+        }
+        options.urAccepted = acceptUsageReporting ? mUrVersionMax : Options.USAGE_REPORTING_DENIED;
+        mConfig.options = options;
```

Comment:
This is weird, why is this setting some mConfig.options as a side effect?

Reply:
I can't see a save or anything anywhere.

Reply:
see MainActivity#showUsageReportingDialog for restApi.restart();
Maybe we should rename the restApi.restart() function to restApi.saveAndRestart() in a follow-up PR?

Reply:
For improvement on this matter, please see PR #1158

Reply:
Still, I don't see a save or anything called.

Reply:
The save is here:
![image](https://user-images.githubusercontent.com/16361913/41512855-edb9c1c6-7291-11e8-8cad-c2ca4ab87565.png)
lines 490 and 494

Reply:
Yeah, let's rename the function in a separate PR.

Reply:
We already did this.